src/Rendering/prePassRenderer.ts

```diff
@@ -377,7 +347,109 @@ export class PrePassRenderer {
+    }
+
+    private _prepareFrame(prePassRenderTarget: PrePassRenderTarget, faceIndex?: number, layer?: number) {
+        if (prePassRenderTarget.renderTargetTexture) {
+            prePassRenderTarget.renderTargetTexture._prepareFrame(this._scene, faceIndex, layer, prePassRenderTarget.renderTargetTexture.useCameraPostProcesses);
+        } else if (this._postProcessesSourceForThisPass.length) {
+            this._scene.postProcessManager._prepareFrame();
+        } else {
+            this._engine.restoreDefaultFramebuffer();
+        }
+    }
+
+    private _renderPostProcesses(prePassRenderTarget: PrePassRenderTarget, faceIndex?: number) {
+        const firstPP = this._postProcessesSourceForThisPass[0];
+        let outputTexture = firstPP ? firstPP.inputTexture : (prePassRenderTarget.renderTargetTexture ? prePassRenderTarget.renderTargetTexture.getInternalTexture() : null);
+
+        // Build post process chain for this prepass post draw
+        let postProcessChain = this._currentTarget._beforeCompositionPostProcesses;
+
+        if (this._needsCompositionForThisPass) {
+            postProcessChain = postProcessChain.concat([this._currentTarget.imageProcessingPostProcess]);
+        }
+
+        // Activates and renders the chain
+        if (postProcessChain.length) {
+            this._scene.postProcessManager._prepareFrame(this._currentTarget.getInternalTexture()!, postProcessChain);
+            this._scene.postProcessManager.directRender(postProcessChain, outputTexture, false, faceIndex);
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public _afterDraw(faceIndex?: number, layer?: number) {
+        if (this._enabled && this._currentTarget.enabled) {
+            this._prepareFrame(this._currentTarget, faceIndex, layer);
+            this._renderPostProcesses(this._currentTarget, faceIndex);
+        }
+    }
+
+    /**
+     * Clears the current prepass render target (in the sense of settings pixels to the scene clear color value)
+     * @hidden
+     */
+    public _clear() {
+        if (this._enabled && this._currentTarget.enabled) {
+            this._bindFrameBuffer(this._currentTarget);
+
+            // Clearing other attachment with 0 on all other attachments
+            this._engine.bindAttachments(this._clearAttachments);
+            this._engine.clear(this._clearColor, true, false, false);
+            // Regular clear color with the scene clear color of the 1st attachment
+            this._engine.bindAttachments(this._defaultAttachments);
+        }
+    }
+
+    private _bindFrameBuffer(prePassRenderTarget: PrePassRenderTarget) {
+        if (this._enabled && this._currentTarget.enabled) {
+            this._currentTarget._checkSize();
+            var internalTexture = this._currentTarget.getInternalTexture();
+            if (internalTexture) {
+                this._engine.bindFramebuffer(internalTexture);
+            }
+        }
+    }
+
+    private _setState(enabled: boolean) {

```

Comment:
Maybe `_setEnabled` would be clearer? Or do you plan to add other states than simply `enabled`?